lib/ansible/executor/task_executor.py

```diff
@@ -571,10 +571,13 @@ def _evaluate_failed_when_result(result):
                     vars_copy.update(result['ansible_facts'])
                 vars_copy.update({'ansible_facts': result['ansible_facts']})
 
-            # set the failed property if the result has a non-zero rc. This will be
-            # overridden below if the failed_when property is set
-            if result.get('rc', 0) != 0:
-                result['failed'] = True
```

Comment:
Correct, but that is the whole point of the PR.

For Windows you have non-zero return codes that are not errors. And this little statement caused them to be interpreted as a failure. The module did return failed=False, nevertheless Ansible looks at the rc value.